pandas/tests/test_categorical.py

```diff
@@ -57,6 +57,13 @@ def test_getitem_listlike(self):
         expected = c[np.array([100000]).astype(np.int64)].codes
         tm.assert_numpy_array_equal(result, expected)
 
+    def test_getname_category(self):
+        result = 'A'
+        s = pd.Series([1, 2, 3], name='A').astype('category')
+        s = s.cat.set_categories([1, 2, 3])
+        expected = s.astype('category').name
+        tm.assert_almost_equal(result, expected)
+
```

Comment:
Can you add additional tests for the other delegated methods that this should apply to? `set_ordered`, etc?